### ISSUE TITLE
Port core concepts of Netflix Hystrix

### ISSUE THREAD

**Issue by rogeralsing** [rogeralsing/Microphone#15]
Explore https://github.com/Netflix/Hystrix to see if we can port some of the core concepts for reciliency.

**Comment by madhon** [comment]
What about using  [Polly](https://github.com/michael-wolfenden/Polly) for part of it ?

**Comment by rogeralsing** [comment]
Yes, I've been watching Polly and think it might be at least part of the solution.

I do find the Polly Circuit breaker somewhat odd as it doesn't take errors over time into account, any successful call resets the breaker.
And I'd also like to see some sort of integration with the error handling and monitoring, not sure if Polly allows you to intercept everything

**Comment by madhon** [comment]
OK, there is an old port of Hystrix to .net on codeplex that could be worth checking for ideas as well

**Comment by bhugot** [comment]
https://github.com/Travix-International/Hystrix.Dotnet/
